shell/common/platform_util.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 Slack Technologies, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/common/platform_util.h"
+
+#include <utility>
+
+#include "base/bind.h"
+#include "base/task/thread_pool.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+#include "shell/common/platform_util_internal.h"
+
+namespace platform_util {
+
+void TrashItemOnBlockingThread(
+    const base::FilePath& full_path,
+    base::OnceCallback<void(bool, const std::string&)> callback) {
+  std::string error;
+  bool success = internal::PlatformTrashItem(full_path, &error);
+  content::GetUIThreadTaskRunner({})->PostTask(
+      FROM_HERE, base::BindOnce(std::move(callback), success, error));
+}
+
+void TrashItem(const base::FilePath& full_path,
+               base::OnceCallback<void(bool, const std::string&)> callback) {
+  // XXX: is continue_on_shutdown right?
+  base::ThreadPool::PostTask(FROM_HERE,
+                             {base::MayBlock(), base::WithBaseSyncPrimitives(),
+                              base::TaskPriority::USER_BLOCKING,
```

Comment:
why can't this be `USER_VISIBLE` instead ? given the api being async, are we really required to respond the user interaction immediately ?

Reply:
I'm not sure what the distinction is here technically, but Chromium uses this for its "OpenItem" calls: https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/platform_util.cc;l=69;drc=14bd4f1ace32db87ec49e6c535f7910f71f1d89a

Reply:
You are right, I thought there would be a priority difference based on the definition [here](https://source.chromium.org/chromium/chromium/src/+/master:base/task/task_traits.h;l=63), but seems like they have same priority https://source.chromium.org/chromium/chromium/src/+/master:base/task/thread_pool/task_tracker.cc;l=480